sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -990,6 +990,29 @@ class DataSourceV2SQLSuite
     assert(catalogManager.currentNamespace === Array("ns1", "ns2"))
   }
 
+  test("ShowCurrentNamespace: basic tests") {
+    def testShowCurrentNamespace(expectedCatalogName: String, expectedNamespace: String): Unit = {
+      val schema = new StructType()
+        .add("catalog", StringType, nullable = false)
+        .add("namespace", StringType, nullable = false)
+      val df = sql("SHOW CURRENT NAMESPACE")
+      val rows = df.collect
+
+      assert(df.schema === schema)
+      assert(rows.length == 1)
+      assert(rows(0).getAs[String](0) === expectedCatalogName)
+      assert(rows(0).getAs[String](1) === expectedNamespace)
+    }
+
+    // Initially, the v2 session catalog is set as a current catalog.
+    testShowCurrentNamespace("spark_catalog", "default")
+
+    sql("Use testcat")
+    testShowCurrentNamespace("testcat", "")
+    sql("Use testcat.ns1.ns2")
+    testShowCurrentNamespace("testcat", "ns1.ns2")
```

Comment:
I guess we need to avoid the side effect of `USE` command at this end of this test case. Or do you want to add `afterEach` in this test suite?

Reply:
Catalog states are cleaned up after each test class inside `after`. So the current set up should be enough. @cloud-fan please correct me if I am wrong here.

Reply:
Got it. I missed that during check this part only. Thanks.